health-mate-client/src/api/genericApi.ts

```diff
@@ -0,0 +1,65 @@
+import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
+import { RestMethod } from "../shared/types/RestMethod";
+import { client } from "../shared/helpers/client";
+
+type SuccessResponse<T> = { status: 'success'; data: T };
+type ErrorResponse = { status: 'error'; error: string };
+type ApiResponse<T> = SuccessResponse<T> | ErrorResponse;
+
+interface RequestParams<T> {
+    method: RestMethod;
+    url: string;
+    data?: T;
+    config?: AxiosRequestConfig;
+}
+
+export async function apiRequest<RequestData, ResponseData>(
+    params: RequestParams<RequestData>
+): Promise<ApiResponse<ResponseData>> {
+
+    function restMethodToString(method: RestMethod): string {
```

Comment:
Please, remove it...
to get a string values of enum use this RestMethod[RestMethod.Get]